extensions/Needinfo/Extension.pm

```diff
@@ -172,6 +172,10 @@ sub bug_start_of_update {
                 if ($requestee ne 'anyone') {
                     _check_requestee($requestee);
                     $needinfo_flag->{requestee} = $requestee;
+                    my $requestee_obj = Bugzilla::User->check($requestee);
+                    if ($requestee_obj && !$requestee_obj->can_see_bug($bug->id)) {
```

Comment:
nit: no need to check $requestee_obj; ->check() will throw an exception if it's bad.